augur/tasks/github/traffic/tasks.py

```diff
@@ -0,0 +1,76 @@
+import time
+import logging
+
+from augur.tasks.init.celery_app import celery_app as celery, engine
+from augur.application.db.data_parse import *
+from augur.tasks.github.util.github_paginator import GithubPaginator
+from augur.tasks.github.util.github_task_session import GithubTaskSession
+from augur.tasks.util.worker_util import remove_duplicate_dicts
+from augur.tasks.github.util.util import get_owner_repo
+from augur.application.db.models import RepoClone, Repo
+from augur.application.db.util import execute_session_query
+
+@celery.task
+def collect_github_repo_clones_data(repo_git: str) -> None:
+    
+    logger = logging.getLogger(collect_github_repo_clones_data.__name__)
+
+    # using GithubTaskSession to get our repo_obj for which we will store data of clones
+    with GithubTaskSession(logger) as session:
+
+        query = session.query(Repo).filter(Repo.repo_git == repo_git)
+        repo_obj = execute_session_query(query, 'one')
+        repo_id = repo_obj.repo_id
+
+        owner, repo = get_owner_repo(repo_git)
+
+        logger.info(f"Collecting Github repository clone data for {owner}/{repo}")
+    
+    clones_data = retrieve_all_clones_data(repo_git, logger)
+
+    if clones_data:
+        process_clones_data(clones_data, f"{owner}/{repo}: Traffic task", repo_id, logger)
+    else:
+        logger.info(f"{owner}/{repo} has no clones")
+
+
```

Comment:
retrieve_all_clones_data and process_clones_data should be in the core.py file of the traffic directory ideally just for organizational reasons. Although, it should still work without these being in a separate file.

Reply:
I have no strong opinions here. I think this person probably followed a random task example they encountered.